### ISSUE TITLE
parse_header_git() cannot parse patch file created by core git containing only one renamed file (100% similarity, no content modifications)

### ISSUE THREAD

**Issue by milke** [libgit2/libgit2#4322]
### Reproduction steps
* Rename just one file in a repository, without modifying its content. Stage and commit. Create a patch file from that commit **using core git (not using the libgit2 library)**.
* Try to parse the patch file, for example, with git_diff_from_buffer().

### Expected behavior
The patch file is parsed and git_diff structure successfully created.

### Actual behavior
git_diff_from_buffer() fails (the failure actually happens in parse_header_git()) with error **invalid patch header at line x**.

### Version of libgit2 (release number or SHA1)
0.26.0

### Operating system(s) tested
macOS 10.12.5

### Additional comment
Interestingly, this is not reproducible with patch files created with the library. Generally, patch file in such case created by core git contains only information about renaming, like:

```
---
 old_name.c => new_name.c | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 rename old_name.c => new_name.c  (100%)

diff --git a/old_name.c b/new_name.c
similarity index 100%
rename from old_name.c
rename to new_name.c
-- 
2.9.3
```

Patch file created by the library looks like a a file with the old name is deleted and a new file with the new name is added:

```
---
 new_name.c        | 38 ++++++++++++++++++++++++++++++++++++++
 old_name.c | 38 --------------------------------------
 2 files changed, 38 insertions(+), 38 deletions(-)
 create mode 100644 new_name.c
 delete mode 100644 old_name.c

diff --git a/new_name.c b/new_name.c
new file mode 100644
index 0000000..9bbcdc9
--- /dev/null
+++ b/new_name.c
@@ -0,0 +1,38 @@
... <here diff info for added new file>...
diff --git a/old_name.c b/old_name.c
deleted file mode 100644
index 9bbcdc9..0000000
--- a/old_name.c
+++ /dev/null
@@ -1,38 +0,0 @@
... <here diff info for deleted old file>...
--
libgit2 0.26.0
```

**Comment by pks-t** [comment]
Thanks for the report. The issue results from us not recognizing "-- " as a patch terminator. For our own generated diff, there is a textual diff previous to the terminating "-- ", so this becomes a non-issue there. I'll fix the issue as part of #4308, which already refactors our patch parsing algorithm.

**Comment by ethomson** [comment]
Fixed!
